src/mongocxx/collection.cpp

```diff
@@ -65,6 +65,23 @@ stdx::string_view collection::name() const noexcept {
     return _impl->name;
 }
 
+void collection::name(stdx::string_view new_name, bool overwrite_old) {
+  bson_error_t error;
+
+  _impl->name = (std::string)new_name;
+  auto result = mongoc_collection_rename(
+      _impl->collection_t
+    , _impl->database_name.c_str()
+    , _impl->name.c_str()
```

Comment:
Sure about that?

https://mnmlstc.github.io/core/string.html#basic_string_view:T:::dataCCE

"This pointer is not guaranteed to be null terminated, and should be treated as such."